segway/input_master.py

```diff
@@ -139,201 +140,253 @@ def make_zero_diagonal_table(length):
     return res
 
 
-def format_indexed_strs(fmt, num):
-    full_fmt = fmt + "%d"
-    return [full_fmt % index for index in range(num)]
-
-
-def jitter_cell(cell, random_state):
-    """
-    adds some random noise
-    """
-    # get the binary exponent and subtract JITTER_ORDERS_MAGNITUDE
-    # e.g. 3 * 2**10 --> 1 * 2**5
-    max_noise = ldexp(1, frexp(cell)[1] - JITTER_ORDERS_MAGNITUDE)
-
-    return cell + random_state.uniform(-max_noise, max_noise)
-
-jitter = vectorize(jitter_cell)
-
-
 class ParamSpec(object):
     """
     base class for parameter specifications used in input.master files
     """
     type_name = None
     object_tmpl = None
     copy_attrs = ["distribution", "mins", "num_segs", "num_subsegs",
-                  "num_track_groups", "track_groups", "num_mix_components"]
+                  "num_track_groups", "track_groups", "num_mix_components",
+                  "means", "vars", "num_mix_components", "random_state",
+                  "tracks", "resolution", "card_seg_countdown", "seg_table",
+                    "seg_countdowns_initial", "len_seg_strength", "num_bases"]
+
+    jitter_std_bound = 0.2
+    track_names = []
 
     def __init__(self, saver):
         # copy all variables from saver that it copied from Runner
         # XXX: override in subclasses to only copy subset
         copy_attrs(saver, self, self.copy_attrs)
+        self.track_names = []
+        for track in self.tracks:
+            self.track_names.append(track.name)
```

Comment:
Might also be useful to get `self.num_tracks` or something here since you constantly use `len(names)` or some variant later